### PR TITLE
Add an utility fn to strip http(s) out of an URL

### PR DESCRIPTION
**Pull request by lamosty**
I think it's a good addition to Calypso as I've seen .replace( /^https?:\/\//, '' ) on too many places.

cc @gwwar 

Test live: https://calypso.live/?branch=add/without-http-utility-fn

### CODE REVIEW

File: client/lib/site/utils.js [hunk 1]
```diff
@@ -172,5 +172,18 @@ export default {
...
+	 * Returns the supplied URL without the initial http(s).
+	 * @param  {String}  url The URL to remove http(s) from
+	 * @return {?String}     URL without the initial http(s)
+	 */
+	withoutHttp( url ) {
+		if ( ! url ) {
+			return null;
+		}
+
+		return url.replace( /^https?:\/\//, '' );
```
Comment (gwwar) [inline client/lib/site/utils.js]: If our regexes are static, I prefer defining them outside of the calling function so they don't need to be created each time.

**Comment by gwwar** [comment]
This helper method feels like it doesn't belong on the site util and instead on a url util. Could you also update the PR with a few examples you wanted to clean up.

**Comment by ockham** [comment]
> This helper method feels like it doesn't belong on the site util and instead on a url util. 

Yep, like `lib/url`.

**Comment by ockham** [comment]
Also, for best practice -- maybe include a test? :-D

**Comment by lamosty** [comment]
Thanks for the reviews! I moved the fn into `lib/url`, used a Regex literal for better perf, replaced **all** occurrences of `/^https?:\/\//` with the fn (manually) and finally, added a few tests.

Could you give it another look, please?

**Comment by gwwar** [comment]
Looking at the example tests, are we using this to get the host? For example should the following `http://myname.com?foo=bar#anchor` return `myname.com` or `myname.com?foo=bar#anchor`?

If it's the first, we can do:
```javascript
import { parse } from 'url';
const host = parse( myExampleUrl ).host;
```

**Comment by lamosty** [comment]
> Looking at the example tests, are we using this to get the host? 

Uhh, I overlooked the fact that we have the `url` lib in codebase. My motivation for creating this function was to replace all the occurrences of `.replace( /^https?:\/\//, '' )` with it. I think that it doesn't return the host, it simply removes the initial `http(s)://` from an URL. In many cases, that produces a host. But I think it should also return `myname.com?foo=bar#anchor` if you pass `http://myname.com?foo=bar#anchor` to it.

I'm going to add your example to the tests so it's more clear what it does.

File: client/lib/url/index.js [hunk 2]
```diff
@@ -26,8 +26,24 @@ function isHttps( url ) {
...
 
+const urlWithoutHttpRegex = /^https?:\/\//;
+
+/**
+ * Returns the supplied URL without the initial http(s).
+ * @param  {String}  url The URL to remove http(s) from
+ * @return {?String}     URL without the initial http(s)
+ */
+function withoutHttp( url ) {
+	if ( ! url ) {
```
Comment (gwwar) [inline client/lib/url/index.js]: Empty strings will be falsey. It may be safer to return `url` in this case if you look at where this is being called.

**Comment by gwwar** [comment]
👍 Code here looks good. Feel free to 🚢 when you're ready.

File: client/lib/url/index.js (hunk 2 above)
Comment (lamosty) [inline client/lib/url/index.js]: Oh, agreed. Great catch. Will fix!

**Comment by lamosty** [comment]
Fixed the empty string issue. Going to merge. Thanks for the review!

**Pull request merged**